packages/@azure/servicebus/data-plane/lib/clientEntityContext.ts

```diff
@@ -113,14 +118,23 @@ export namespace ClientEntityContext {
       entityPath: entityPath,
       requestResponseLockedMessages: new ConcurrentExpiringMap<string>(),
       isSessionEnabled: !!options.isSessionEnabled,
-      messageSessions: {}
+      messageSessions: {},
+      expiredMessageSessions: {}
     };
 
     (entityContext as ClientEntityContext).sessionManager = new SessionManager(
       entityContext as ClientEntityContext
     );
 
     (entityContext as ClientEntityContext).getReceiver = (name: string, sessionId?: string) => {
+      if (sessionId != undefined && entityContext.expiredMessageSessions[sessionId] === true) {
+        // TODO: Use an error constructor utility which could also serve as interface documentation for all types of errors thrown
+        const error = new Error();
+        error.name = "SessionLockLostError";
+        error.message = `Session Lock is lost for: ${sessionId}. Open a new Session using getSessionReceiver()`;
```

Comment:
I believe the message can be passed to the `Error` constructor

Reply:
Sure, can revisit this as part of #1145 maybe